lighthouse-core/gather/gatherers/fonts.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer');
+const fontFaceDescriptors = [
+  'font-family',
+  'font-style',
+  'font-weight',
+  'font-stretch',
+  'unicode-range',
+  'font-variant',
+  'font-feature-settings',
+];
+
+/* eslint-disable */
+function getAllLoadedFonts() {
+  const getFont = fontFace => {
+    const fontRule = {};
+    fontFaceDescriptors.forEach(descriptor => {
+      const descriptorWithoutFont = descriptor.replace('font-', '');
+      fontRule[descriptorWithoutFont] = fontFace[descriptorWithoutFont];
+    });
+
+    return fontRule;
+  };
+
+  return document.fonts.ready.then(() => {
+    return Array.from(document.fonts).filter(fontFace => fontFace.status === 'loaded')
+      .map(getFont);
+  });
+}
+
+function getFontFaceFromStylesheets() {
+  function resolveUrl(url) {
+    const link = document.createElement('a');
+    link.href = url;
+
+    return link.href;
+  }
+
+  function getSheetsFontFaces(stylesheet) {
+    const fontFaceRules = [];
+    if (stylesheet.cssRules) {
+      for (const rule of stylesheet.cssRules) {
+        if (rule instanceof CSSFontFaceRule) {
+          const fontsObject = {
+            display: rule.style.fontDisplay || 'auto',
+            family: rule.style.fontFamily.replace(/"|'/g, ''),
+            stretch: rule.style.fontStretch || 'normal',
+            style: rule.style.fontStyle || 'normal',
+            weight: rule.style.fontWeight || 'normal',
```

Comment:
from https://github.com/GoogleChrome/lighthouse/pull/3831/files/bf0d8e3cc3c9a16b93ff26b59048563193db85be#r159987426 i think you still need to collect `unicodeRange` `fontVariant` and `fontFeatureSettings` from the `rule.style` as well.